jOOL-java-8/src/main/java/org/jooq/lambda/tuple/Range.java

```diff
@@ -35,18 +35,23 @@ public Range(Tuple2<T, T> tuple) {
     }
 
     private static <T extends Comparable<T>> Tuple2<T, T> r(T t1, T t2) {
-        return t1.compareTo(t2) <= 0 ? new Tuple2<>(t1, t2) : new Tuple2<>(t2, t1);
+        if (t1 != null && t2 != null)
+            return t1.compareTo(t2) <= 0 ? new Tuple2<>(t1, t2) : new Tuple2<>(t2, t1);
+        else
+            return new Tuple2<>(t1, t2);
```

Comment:
The fact that this is necessary raises more questions: Is swapping input bounds still a valid thing in the presence of possibly unbounded ranges? I think it isn't. But if we want to stay backwards compatible, then this logic is definitely correct...

I've created an issue for this, to keep this in mind: #352